mypyc/test-data/run-dicts.test

```diff
@@ -217,3 +217,13 @@ def test_dict_copy() -> None:
     dd['a'] = 1
     assert dd.copy() == dd
     assert isinstance(dd.copy(), defaultdict)
+
+def test_dict_setdefault() -> None:
+    d: Dict[str, int] = {'a': 1, 'b': 2}
+    assert d.setdefault('a', 2) == 1
+    assert d.setdefault('c', 3) == 3
+    assert d['a'] == 1
+    assert d['c'] == 3
+    assert d.setdefault('a') == 1
+    assert d.setdefault('e') == None
+    assert d.setdefault('e', 100) == None
```

Comment:
Also test a subclass of `dict` that overrides `setdefault`. We should use the overridden method (example where this probably matters: `OrderedDict`). You can create a `dict` subclass that overrides `setdefault` that behaves differently and use that in the test. If this doesn't work correctly right now, you'll probably need to add a helper similar to `CPyDict_KeysView` that has a generic fallback implementation for `dict` subclasses.